include/mbedtls/ssl.h

```diff
@@ -382,6 +386,8 @@ union mbedtls_ssl_premaster_secret
 
 #define MBEDTLS_PREMASTER_SIZE     sizeof( union mbedtls_ssl_premaster_secret )
 
+#define MBEDTLS_SSL_CHK_BUF(cur, end, need) ( end < cur || (size_t)( end - cur ) < (size_t) need )
```

Comment:
This name/return value does not match our usual conventions. Functions (or in that case, function-like macros) that have "check" or equivalent in their name are supposed to return (or evaluate to) 0 in case the thing checked was found to be acceptable. See "return type" in <https://tls.mbed.org/kb/development/mbedtls-coding-standards>

In the past people have been bitten by inconsistent conventions about return value for check_stuff() (one vulnerability in X509 verification in another library was associated to this IIRC), so let's try to avoid similar issues and have consistently

    if( mbedtls_chk_stuff(foo, bar) != 0 )
    {
        // handle error
    }

More as a matter of personal taste, why not a static inline function?